### ISSUE TITLE
Add new population dataset

### ISSUE THREAD

**Issue by pz-max** [pz-max/pypsa-africa#174]
Some new data population `raster` data was released recently. See more:
https://www.linkedin.com/posts/andy-tatem-b199b018_worldpop-and-esri-are-pleased-to-announce-activity-6873913543029350401-RRtp

**Use:**
We use such rasterized data and fill that into arbitrary .geojson shapes i.e. of a country, subregions or similar. This data is and might be used for different purposes in the workflow, for instance, predicting the demand or distributing the cluster nodes across countries.

**TODO:** (suitable for new contributors)
Creating a pull request to add this data is highly welcome. Feel free to reach out to take this challenge. It will impact `build_shapes.py` and `cluster_network.py` and need to be tested properly.

**Comment by Tomkourou** [comment]
You can assign me to this one :)

**Comment by davide-f** [comment]
Thank you for the contribution, feel free to comment if you need help

**Comment by Tomkourou** [comment]
So we want to integrate the new _higher resolution_ population count from worldpop right? 

A typical country would look like this -> https://www.worldpop.org/geodata/summary?id=49737

With the associated tiff url -> https://data.worldpop.org/GIS/Population/Global_2000_2020_Constrained/2020/BSGM/DZA/dza_ppp_2020_UNadj_constrained.tif

I haven't found constrained data files, is that a problem? 

Also I noticed they have an API which might be nicer to use for updates/new datasets. Any reason why you opted not to use it?

**Comment by pz-max** [comment]
Hi @Tomkourou, great to see you giving this a try :)
The worldpop data is currently downloaded here:
https://github.com/pypsa-meets-africa/pypsa-africa/blob/338fe3614c7e757af5eada5e9c8ec1bb137adbf9/scripts/build_shapes.py#L305-L363

Looking at the population data which @davide-f used from:
https://data.worldpop.org/GIS/Population/ I can see that there was already a 1km resolution available when the data was released in 2020. The question I am having now is, what resolution is our data (already 1x1km?)? This means maybe let's check if the dataset is new or they sold old stuff and the real new thing from the LinkedIn post was the ArcGISOnline's Living Atlas integration.

Regarding API, it might be indeed smoother to get updates but requires a bit more time to set it up. It seems 1000 API requests without API keys are plenty enough: https://www.worldpop.org/sdi/api_rate_limits In case, you are motivated to integrate the API, I would probably recommend creating a new rule `download_population` where you use the API as default & if it fails the static hosted data. The outputs of the rule are then used at build shapes.

@davide-f , what are your thoughts?

**Comment by davide-f** [comment]
We should be using the 100x100 m (5 arc-second) resolution; we could verify what different it makes to use 1km (30 arc-second) cells instead in terms of quantifying the population for the same shapes versus computational time.

That problem is really linked to the last open point in https://github.com/pypsa-meets-africa/pypsa-africa/issues/56 that aims at improving the GPD estimate for the region.
The GDP raster we are using is 5 arc, which is way larger than those used for population and that leads to errors in the estimation; disaggregation techniques using population data may for example improve that.

**Comment by Tomkourou** [comment]
I have opened a PR for this issue [here ](https://github.com/pypsa-meets-africa/pypsa-africa/pull/325). So far looking at the old implementation I don't see tif files by country for 100mx100m resolution. The only updated files are here https://data.worldpop.org/GIS/Population/Single_country/UKR/v1/constrained/100m/. However when accessing the API it's possible to get the 100mx100m constrained files by country. 

So far I added a function which as close as possible resembles the one you already had but makes a request to the API instead. Let me know what you think.
